internal/sqlquery/scraper.go

```diff
@@ -107,3 +109,142 @@ func (s *Scraper) Shutdown(_ context.Context) error {
 	}
 	return nil
 }
+
+func BuildDataSourceString(driver string, dataSource DataSourceConfig) (string, error) {
+	switch driver {
+	case "postgres":
+		return buildPostgreSQLString(dataSource)
+	case "mysql":
+		return buildMySQLString(dataSource)
+	case "snowflake":
+		return buildSnowflakeString(dataSource)
+	case "sqlserver":
+		return buildSQLServerString(dataSource)
+	case "oracle":
+		return buildOracleString(dataSource)
+	default:
+		return "", fmt.Errorf("unsupported driver: %s", driver)
+	}
+}
+
+func buildPostgreSQLString(conn DataSourceConfig) (string, error) {
+	// PostgreSQL connection string format: postgresql://user:pass@host:port/db?param1=value1&param2=value2
+	var auth string
+	if conn.Username != "" {
+		auth = fmt.Sprintf("%s:%s@", url.QueryEscape(conn.Username), url.QueryEscape(string(conn.Password)))
```

Comment:
This is a bit of a nit, but why the `@` at the end of the `auth` string? Why not just add it in the later step of compiling parts of the connection string?

Same question for other connection strings. I think it makes sense to have valid building blocks first, then when they're being put together include the separator logic.